metricbeat/module/kvm/status/status.go

```diff
@@ -0,0 +1,171 @@
+	// limit is defined by REMOTE_DOMAIN_MEMORY_STATS_MAX
+	// based on https://github.com/libvirt/libvirt/blob/5bb07527c11a6123e044a5dfc48bdeccee144994/src/remote/remote_protocol.x#L136
+	maximumStats = 11
+	// flag VIR_DOMAIN_AFFECT_CURRENT passed to collect memory stats
+	// based on https://libvirt.org/html/libvirt-libvirt-domain.html#virDomainModificationImpact
+	flags = 0
+)
+
+// init registers the MetricSet with the central registry as soon as the program
+// starts. The New function will be called later to instantiate an instance of
+// the MetricSet for each host defined in the module's configuration. After the
+// MetricSet has been created then Fetch will begin to be called periodically.
+func init() {
+	mb.Registry.MustAddMetricSet("kvm", "status", New,
+		mb.DefaultMetricSet(),
+	)
+}
+
+// MetricSet holds any configuration or state information. It must implement
+// the mb.MetricSet interface. And this is best achieved by embedding
+// mb.BaseMetricSet because it implements all of the required mb.MetricSet
+// interface methods except for Fetch.
+type MetricSet struct {
+	mb.BaseMetricSet
+	Timeout time.Duration
+	HostURL *url.URL
+}
+
+// New creates a new instance of the MetricSet. New is responsible for unpacking
+// any MetricSet specific configuration options if there are any.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	cfgwarn.Beta("The kvm status metricset is beta.")
+	u, err := url.Parse(base.HostData().URI)
+	if err != nil {
+		return nil, err
+	}
+
+	return &MetricSet{
+		BaseMetricSet: base,
+		Timeout:       base.Module().Config().Timeout,
+		HostURL:       u,
+	}, nil
+}
+
+// Fetch methods implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(report mb.ReporterV2) error {
+	var (
+		c   net.Conn
+		err error
+	)
+
+	u := m.HostURL
+
+	if u.Scheme == "test" {
+		// when running tests, a mock Libvirt server is used
+		c = libvirttest.New()
+	} else {
+		address := u.Host
+		if u.Host == "" {
+			address = u.Path
+		}
+
+		c, err = net.DialTimeout(u.Scheme, address, m.Timeout)
+		if err != nil {
+			return errors.Wrapf(err, "cannot connect to %v", u)
+		}
+	}
+
+	defer c.Close()
+
+	l := libvirt.New(c)
+	if err = l.Connect(); err != nil {
+		return errors.Wrap(err, "error connecting to libvirtd")
+	}
+	defer func() {
+		if err = l.Disconnect(); err != nil {
+			msg := errors.Wrap(err, "failed to disconnect")
+			report.Error(msg)
+			m.Logger().Error(msg)
+		}
+	}()
+
+	domains, err := l.Domains()
+	if err != nil {
+		return errors.Wrap(err, "error listing domains")
+	}
+
+	for _, d := range domains {
+
+		state, err := l.DomainState(d.Name)
+		if err != nil {
+			continue
+		}
+		reported := report.Event(mb.Event{
+			MetricSetFields: common.MapStr{
+				"id":   d.ID,
+				"name": d.Name,
```

Comment:
id and name could be moved to some field at the module level, but let's keep this for a future change because this should be also changed in the other metricset.
```suggestion
			ModuleFields: common.MapStr{
				"id":   d.ID,
				"name": d.Name,
			},
			MetricSetFields: common.MapStr{
```